src/web/server/document.tsx

```diff
@@ -55,6 +55,79 @@ export const document = ({ data }: Props) => {
 		),
 	);
 
+	const loadableExtractor = new ChunkExtractor({
+		stats: loadableManifestJson,
+		entrypoints: ['react'],
+	});
+
+	// The lodable-components docs want us to use extractor.collectChunks() but
+	// we don't have the traditional same <App /> rendered on server and client.
+	// Our data structure is vastly different (CAPIType vs CAPIBrowserType) and
+	// our architecture expects src/web/App to be client-side only, therefore
+	// it is difficult for us to use collectChunks to automatically extract the
+	// component splits that we want.
+	// However, this does actually suit our architecture as we can use the CAPI
+	// component reference.
+	const allChunks: LoadableComponents = [
+		{ chunkName: 'EditionDropdown', addWhen: 'always' },
+		{
+			chunkName: 'elements-YoutubeBlockComponent',
+			addWhen: 'model.dotcomrendering.pageElements.YoutubeBlockElement',
+		},
+		{
+			chunkName: 'elements-RichLinkComponent',
+			addWhen: 'model.dotcomrendering.pageElements.RichLinkBlockElement',
+		},
+	];
+	// We want to only insert script tags for the elements or main media elements on this page view
+	// so we need to check what elements we have and use the mapping to the the chunk name
+	const CAPIElements: CAPIElement[] = CAPI.blocks[0]
+		? CAPI.blocks[0].elements
+		: [];
+	const { mainMediaElements } = CAPI;
+	// Filter the chunks defined above by whether
+	// the 'addWhen' value is 'always' or matches
+	// any elements in the body or main media element
+	// arrays for the page request.
+	const chunksForPage: (
+		| EditionDropdownLoadable
+		| YoutubeBlockLoadable
+		| RichLinkBlockLoadable
+	)[] = allChunks.filter((chunk) =>
+		[...CAPIElements, ...mainMediaElements].some(
+			(block) =>
+				chunk.addWhen === 'always' || block._type === chunk.addWhen,
+		),
+	);
+	// Once we have the chunks for the page, we can add them directly to the loadableExtractor
+	chunksForPage.forEach((chunk) => {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+		// @ts-ignore
+		loadableExtractor.addChunk(chunk.chunkName); // addChunk is *undocumented* and not in TS types. It allows manually adding chunks to extractor.
```

Comment:
bit odd that's it's undocumented - not worried that it might get removed in a future release?

Reply:
Maybe, but it's not currently possible to use it without, so if a future release deprecates it then we'd have to rethink anyway.